src/relax/ir/binding_rewrite.cc

```diff
@@ -218,13 +218,15 @@ class RemoveUnusedVars : public ExprMutator {
             for (size_t i = prev_size; i < unused.size(); ++i) {
               users.erase(unused[i]);
               // remove def site.
+              auto updated = Map<Var, Array<Var>>();
```

Comment:
created separate map in inner loop seems to be quite expensive. can we switch to a different algorithm? e.g. first mark use_count, then do analysis to mark things to be rmeoved, finally remove everything that is marked

Reply:
OK got it